apis/v1alpha1/gatewayclass_types.go

```diff
@@ -62,20 +62,58 @@ type GatewayClassSpec struct {
 	// +kubebuilder:validation:MaxLength=253
 	Controller string `json:"controller"`
 
-	// ParametersRef is a controller-specific resource containing the
-	// configuration parameters corresponding to this class. This is optional if
-	// the controller does not require any additional configuration.
+	// ParametersRef is a reference to a resource that contains the configuration
+	// parameters corresponding to the GatewayClass. This is optional if the
+	// controller does not require any additional configuration.
 	//
-	// Parameters resources are implementation specific custom resources. These
-	// resources must be cluster-scoped.
+	// ParametersRef can reference a standard Kubernetes resource, i.e. ConfigMap,
+	// or an implementation-specific custom resource. The resource can be
+	// cluster-scoped or namespace-scoped.
 	//
 	// If the referent cannot be found, the GatewayClass's "InvalidParameters"
 	// status condition will be true.
 	//
 	// Support: Custom
 	//
 	// +optional
-	ParametersRef *LocalObjectReference `json:"parametersRef,omitempty"`
+	ParametersRef *ParametersReference `json:"parametersRef,omitempty"`
+}
+
+// ParametersReference identifies an API object containing controller-specific
+// configuration resource within the cluster.
+type ParametersReference struct {
+	// Group is the group of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Group string `json:"group"`
+
+	// Kind is kind of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// Scope represents if the referent is a Cluster or Namespace scoped resource.
+	// This may be set to "Cluster" or "Namespace".
+	// +kubebuilder:validation:Enum=Cluster;Namespace
+	// +kubebuilder:default=Cluster
```

Comment:
I think for defaulting to work we need an `omitempty` tag. If we have an `omitempty` tag, we should also mark the field as optional to match the API conventions. @danehans can confirm here, but that's how I remember this working.

Reply:
I tested and this is indeed the case. Updated.